driver/registry_base.go

```diff
@@ -102,11 +104,26 @@ func (m *RegistryBase) OAuth2AwareMiddleware() func(h http.Handler) http.Handler
 	return m.oa2mw
 }
 
+func (m *RegistryBase) AddCORSonHandler(h http.Handler) http.Handler {
+	c := cors.New(cors.Options{
+		AllowedMethods:   []string{"GET", "POST", "PUT", "PATCH", "DELETE"},
+		AllowedHeaders:   []string{"Authorization", "Content-Type"},
+		ExposedHeaders:   []string{"Content-Type"},
+		AllowCredentials: true,
+	})
+	return c.Handler(h)
+}
```

Comment:
I think we should create a new cors instance only once inside the registry base and base it on the config we have available as a `provider` in the `config` pacakge. See https://github.com/ory/hydra/blob/43b391d95f17cfd7414786cec0c602c15e29e956/driver/config/serve.go#L62-L69

Here is an example of how we use it https://github.com/ory/hydra/blob/63402dee7604141118fead91491abe6763150f1c/x/oauth2cors/cors.go#L41-L52